tools/cldr-apps/js/src/views/LoginButton.vue

```diff
@@ -38,48 +60,94 @@ export default {
+        this.logText = "Cancel";
+      } else if (loggedIn) {
         this.logText = "Log Out";
       } else {
         this.logText = "Log In";
       }
     },
-    async loginout() {
-      if (this.loggedIn) {
-        await fetch(`api/auth/logout?session=${cldrStatus.getSessionId()}`);
-        // now reload this page now that we've logged out
-        await window.location.reload();
-      } else {
-        if (this.userName && this.password) {
-          try {
-            const response = await fetch(
-              `api/auth/login?remember=${this.remember}`,
-              {
-                method: "POST",
-                headers: {
-                  "Content-Type": "application/json",
-                  Accept: "application/json",
-                },
-                body: JSON.stringify({
-                  email: this.userName,
-                  password: this.password,
-                }),
-              }
-            );
-            // TODO: if !response.ok, show login failed
-            const logintoken = await (await response).json();
-            if (/*logintoken.user && */ logintoken.sessionId) {
-              // logged in OK.
-              cldrStatus.setSessionId(logintoken.sessionId);
-              run(); // Restart everything
-            } // TODO: else, handle error.
-          } catch (e) {
-            // TODO: popover needs to show that login failed
-            throw e;
+    /**
+     * Log out
+     */
+    async logout() {
+      await fetch(`api/auth/logout?session=${cldrStatus.getSessionId()}`);
+      // now reload this page now that we've logged out
+      await window.location.reload();
+    },
+    /**
+     * Log in (from header button)
+     */
+    async login() {
+      function errBox(message) {
+        console.error("LoginButton.vue: " + message);
+        notification.error({
+          message,
+          placement: "topLeft",
+        });
+      }
+
+      this.loginErrorMessage = null;
+      if (this.userName && this.password) {
+        try {
+          const response = await fetch(
+            `api/auth/login?remember=${this.remember}`,
+            {
+              method: "POST",
+              headers: {
+                "Content-Type": "application/json",
+                Accept: "application/json",
+              },
+              body: JSON.stringify({
+                email: this.userName,
+                password: this.password,
+              }),
+            }
+          );
+          if (!response.ok) {
+            if (response.status == 403) {
+              return errBox("Unauthorized:\nCheck the username and password.");
+            } else {
+              return errBox(`Login failed: HTTP ${response.status}`);
+            }
+            return;
+          }
+          // TODO: if !response.ok, show login failed
+          const logintoken = await (await response).json();
+          if (/*logintoken.user && */ logintoken.sessionId) {
```

Comment:
omit commented-out code?

Reply:
So, "user" was removed by you (thanks!!) to fix the serialization problem. But ultimately, the login endpoint ought to return the user info, and then will use that instead of taking it from the repeated status calls.￼